deploy-operator/controllers/locality_controller.go

```diff
@@ -0,0 +1,222 @@
+// +kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=batch,resources=jobs/status,verbs=get
+
+func (r *LocalityReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	r.ctx = context.Background()
+	r.Log = r.Log.WithValues("locality", req.NamespacedName)
+
+	var locality priov1.Locality
+	if err := r.Get(r.ctx, req.NamespacedName, &locality); err != nil {
+		r.Log.Error(err, "unable to fetch Locality resource")
+
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	ret, err := r.validate(&locality)
+	// Assume the error has been logged and the correct return has been selected
+	if err != nil {
+		return ret, err
+	}
+
+	var cronJobs v1beta1.CronJobList
+
+	if err := r.List(r.ctx, &cronJobs, client.InNamespace(req.Namespace), client.MatchingFields{
+		jobNameField: jobName,
+	}); err != nil {
+		r.Log.Error(err, "unable to fetch list of jobs for controller")
+
+		return ctrl.Result{}, err
+	}
+	r.Log.Info("List of cronjobs returned", "cronjobs", cronJobs, "cronjobs size", len(cronJobs.Items))
+	switch len(cronJobs.Items) {
+	case 0:
+		ret, err = r.scheduleNewJob(&locality)
+		if err != nil {
+			return ret, err
+		}
+	case 1:
+		ret, err = r.validateCurrentJob(&cronJobs.Items[0], &locality)
+		if err != nil {
+			return ret, err
+		}
+
+	default:
+		panic("THIS SHOULD NEVER HAPPEN!")
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *LocalityReconciler) scheduleNewJob(locality *priov1.Locality) (ctrl.Result, error) {
+	cronJob := r.createCronJobTemplate(locality)
+
+	if err := r.Client.Create(r.ctx, &cronJob); err != nil {
+		r.Log.Error(err, "unable to create a new CronJob")
+		return ctrl.Result{}, err
+	}
+
+	job := r.createJobFromCronJob(&cronJob)
+	if err := r.Client.Create(r.ctx, &job); err != nil {
+		r.Log.Error(err, "unable to create an immediate job from the cron job")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+func (r *LocalityReconciler) createJobFromCronJob(cronJob *v1beta1.CronJob) v1.Job {
+	return v1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("init-%s", jobName),
+			Namespace: cronJob.Namespace,
+			Annotations: map[string]string{
+				"cronjob.kubernetes.io/instantiate": "manual",
+			},
+		},
+		Spec: *cronJob.Spec.JobTemplate.Spec.DeepCopy(),
+	}
+}
+
+func (r *LocalityReconciler) createCronJobTemplate(locality *priov1.Locality) v1beta1.CronJob {
+	var jobTTL int32
+	jobTTL = 5
+
+	trueValue := true
+
+	env := []v12.EnvVar{
+		{Name: "KR_ENVIRONMENT_NAME", Value: locality.Spec.EnvironmentName},
+		{Name: "KR_MANIFEST_BUCKET_LOCATION", Value: locality.Spec.ManifestBucketLocation},
+		{Name: "KR_DATA_SHARE_PROCESSORS", Value: strings.Join(locality.Spec.DataShareProcessors, " ")},
+		{Name: "KR_LOCALITY", ValueFrom: &v12.EnvVarSource{
+			FieldRef: &v12.ObjectFieldSelector{
+				FieldPath: "metadata.namespace",
+			},
+		}},
+	}
```

Comment:
I'm not a big fan of making a ConfigMap because that means the operator now needs to keep track of two things for state management, effectively doubling how much work it needs to do. It also means that the operator needs to get extra permissions to read/write ConfigMaps